File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/java/cwe-78/code_before1.java
// Source: Row 2 in ./dataset/CVEfixes/Analysis/results/Java/df_java_cwe_78.xlsx

public void testEscapeSingleQuotesOnArgument()
{
    Shell sh = newShell();

    sh.setWorkingDirectory( "/usr/bin" );
    sh.setExecutable( "chmod" );

    String[] args = { "arg'withquote" };

    List shellCommandLine = sh.getShellCommandLine( args );

    String cli = StringUtils.join( shellCommandLine.iterator(), " " );
    System.out.println( cli );
    assertEquals("cd '/usr/bin' && 'chmod' 'arg'\"'\"'withquote'", shellCommandLine.get(shellCommandLine.size() - 1));
}

File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/python/cwe-78/code_before1.py
# Source: Row 5 in ./dataset/CVEfixes/Analysis/results/Python/df_python_cwe_78.xlsx

@component(AuthenticationProvider)
class OSAuthenticationProvider(AuthenticationProvider):
    id = 'os'
    name = 'OS users'
    allows_sudo_elevation = True

    def authenticate(self, username, password):
        child = None
        try:
            child = pexpect.spawn('/bin/sh', ['-c', '/bin/su -c "/bin/echo SUCCESS" - %s' % username], timeout=5)
            child.expect('.*:')
            child.sendline(password)
            result = child.expect(['su: .*', 'SUCCESS'])
        except Exception as err:
            if child and child.isalive():
                child.close()
            logging.error('Error checking password: %s', err)
            return False
        if result == 0:
            return False
        else:
            return True
        